ablySpec/RealtimeClientChannel.swift

```diff
@@ -13,6 +13,32 @@ class RealtimeClientChannel: QuickSpec {
     override func spec() {
         describe("Channel") {
 
+            // RTL4
+            describe("attach") {
+
+                // RTL4a
+                it("if already ATTACHED or ATTACHING nothing is done") {
+                    let client = ARTRealtime(options: AblyTests.commonAppSetup())
+                    defer { client.close() }
+
+                    var errorInfo: ARTErrorInfo?
+                    let channel = client.channel("test")
+
+                    errorInfo = channel.attach()
+                    expect(errorInfo).to(beNil())
+                    expect(channel.state).to(equal(ARTRealtimeChannelState.Attaching))
+
+                    errorInfo = channel.attach()
+                    expect(errorInfo!.message).to(contain("Already attached"))
```

Comment:
@ricardopereira Well, yes, I can, but this can be fixed right now, by changing this expectation to `expect(errorInfo).to(beNil())`. Then, the API change only would need to change, well, the API, to make it a callback.